racket/collects/racket/private/prop-template-metafunction.rkt

```diff
@@ -0,0 +1,69 @@
+(module prop-template-metafunction '#%kernel
+  (#%provide (protect metafunction
+                      prop:metafunction
+                      (rename prop:metafunction? metafunction?)
+                      metafunction-accessor))
+  
+  ;; The prop:metafunction structure type property can contain an
+  ;; identifier bound to the run-time metafunction procedure, or the index of a
+  ;; field containing such an identifier.
```

Comment:
Could you make it so that `prop:metafunction` can be a function that takes an instance of that structure and produces an identifier? Where you could say
```racket
(struct foo [id]
  #:property prop:metafunction
  (lambda (self) (foo-id self)))
```
Properties like `prop:rename-transformer`, `prop:require-transformer`, `prop:provide-transformer`, `prop:pattern-expander`, and `prop:code-transformer` all allow something like this.

Reply:
Sure, will do!

Reply:
I've started on a commit that does this here on a `prop-template-metafunction-2` branch based on yours:
https://github.com/AlexKnauth/racket/commit/da9e4f06b9b0323c7f5562dbee6b29ac4860ef58